test/manual/kerberos.test.js

```diff
@@ -79,18 +79,73 @@ describe('Kerberos', function () {
     });
   });
 
-  it('validate that CANONICALIZE_HOST_NAME can be passed in', function (done) {
-    if (process.platform === 'darwin') {
-      this.test.skipReason = 'DNS does not resolve with proper CNAME record on evergreen MacOS';
-      this.skip();
-    }
-    const client = new MongoClient(
-      `${krb5Uri}&authMechanismProperties=SERVICE_NAME:mongodb,CANONICALIZE_HOST_NAME:true&maxPoolSize=1`
-    );
-    client.connect(function (err, client) {
-      if (err) return done(err);
-      expect(dns.resolveCname).to.be.calledOnce;
-      verifyKerberosAuthentication(client, done);
+  context('when passing in CANONICALIZE_HOST_NAME', function () {
+    beforeEach(function () {
+      if (process.platform === 'darwin') {
+        this.currentTest.skipReason =
+          'DNS does not resolve with proper CNAME record on evergreen MacOS';
+        this.skip();
+      }
+    });
+
+    context('when the value is true', function () {
```

Comment:
For this entire set of tests, the actual test logic seems to be identical - is there more we can test for each specific test case here to make sure that the code is actually doing something different in each case?

Reply:
I've updated to assert on the dns spy for each of the calls.

Reply:
Could we make the test cases a bit more descriptive in terms of what the spec says should happen? (i.e., for `'none'` and `false`, `"should authenticate without performing any lookups"`, for `'forwardAndReverse'` and `true`, `"should authenticate by performing a forward and reverse lookup"`, for `'forward'`, `"should authenticate by performing a forward DNS lookup only"`). And then probably also add cases for what's expected on error if that's not too difficult to do (if sinon lets us also stub the response from the functions, we could simulate errors there). For the cases that should be identical, we could make sure the logic stays the same by doing something like 
```js
for const option of [true, 'forwardAndReverse'] { 
  context(`when the value is ${option}`, function() {
    it('should authenticate by performing a forward and reverse lookup', ... );
    it('should <error behavior> if the forward lookup fails with an error', ...);
    it('should <error behavior> if the forward lookup fails with no results', ...);
    it('should <error behavior> if the reverse lookup fails with an error', ...);
    it('should <error behavior> if the reverse lookup fails with no results', ...);
  })
}
``` 
and similarly for `[false, 'none']`, with a separate block for `'forward'` only

Reply:
I updated the tests now to test the further error/empty array/success cases on the reverse lookup and the cname lookup. The general case I left as-is since stubbing dns.lookup would affect the other lookup calls and not test this in isolation.